homeassistant/components/camera/usps.py

```diff
@@ -0,0 +1,106 @@
+"""
+Support for a camera made up of usps mail images.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/camera.usps/
+"""
+import os
+import logging
+from datetime import timedelta
+
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.util.dt import utcnow
+from homeassistant.components.usps import DATA_USPS
+from homeassistant.components.camera import Camera
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['usps']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up USPS mail camera."""
+    if discovery_info is None:
+        return
+
+    usps = hass.data[DATA_USPS]
+    add_devices([USPSCamera(hass, usps)])
+
+
+class USPSCamera(Camera):
+    """Representation of the images available from USPS."""
+
+    def __init__(self, hass, usps):
+        """Initialize the USPS camera images."""
+        super(USPSCamera, self).__init__()
+
+        self._hass = hass
+        self._usps = usps
+        self._name = self._usps.name
+        self._session = self._usps.session
+
+        self._mail_img = []
+        self._last_mail = None
+        self._mail_index = 0
+        self._mail_count = 0
+
+        self._timer = None
+
+        # Fetch no mail camera image
+        image_path = os.path.join(
+            os.path.dirname(__file__), 'nomail.jpg')
+        with open(image_path, 'rb') as file:
+            self._no_mail_img = file.read()
+
+    def camera_image(self):
+        """Update the camera's image if it has changed."""
+        self._usps.update()
+        try:
+            self._mail_count = len(self._usps.mail)
+        except TypeError:
+            # No mail
+            return self._no_mail_img
+
+        if self._usps.mail != self._last_mail:
+            # Mail items must have changed
+            self._mail_img = []
+            if self._usps.mail is not None and len(self._usps.mail) >= 1:
+                self._last_mail = self._usps.mail
+                for article in self._usps.mail:
+                    _LOGGER.debug("Fetching article image: %s", article)
+                    img = self._session.get(article['image']).content
+                    self._mail_img.append(img)
+
+        def _interval_update(now):
```

Comment:
This did sound like a great idea, but I'm having issues getting the ```scan_interval``` defined in the camera configuration actually applying to the component.  The default value is respected, but it doesn't seem to pick up the overriding value.

My only code changes are to add a default scan interval:
```python
SCAN_INTERVAL = timedelta(seconds=10)
```
Then remove the existing update functions and add the standard entity functions:
```python
    @property
    def should_poll(self):
        """Update the mail image index periodically."""
        return True

    def update(self):
        """Update mail image index."""
        _LOGGER.debug('Updating usps mail index...')
        if self._mail_index < (self._mail_count - 1):
            self._mail_index += 1
        else:
            self._mail_index = 0
```
I also tried defining ```CONF_SCAN_INTERVAL``` within a PLATFORM_SCHEMA, but it didn't make a difference and I don't think it's necessary.

I'm trying to set the interval with the following in my configuration.yaml:
```yaml
camera:
  - platform: usps
    scan_interval: 5
```